router/utils_test.gno

```diff
@@ -0,0 +1,177 @@
+package router
+
+import (
+	"strings"
+	"testing"
+)
+
+type poolPathWithFeeDivideTestCases struct {
+	name        string
+	input       string
+	wantToken0  string
+	wantToken1  string
+	wantFee     int
+	shouldPanic bool
+}
+
+func TestPoolPathWithFeeDivide(t *testing.T) {
+	tests := []poolPathWithFeeDivideTestCases{
+		{
+			name:        "valid path",
+			input:       "token0:token1:500",
+			wantToken0:  "token0",
+			wantToken1:  "token1",
+			wantFee:     500,
+			shouldPanic: false,
+		},
+		{
+			name:        "valid path with special characters",
+			input:       "r/token_a:r/token_b:3000",
+			wantToken0:  "r/token_a",
+			wantToken1:  "r/token_b",
+			wantFee:     3000,
+			shouldPanic: false,
+		},
+		{
+			name:        "invalid fee format",
+			input:       "token0:token1:abc",
+			shouldPanic: true,
+		},
+		{
+			name:        "missing parts",
+			input:       "token0:token1",
+			shouldPanic: true,
+		},
+		{
+			name:        "empty string",
+			input:       "",
+			shouldPanic: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer func() {
+				r := recover()
+				if (r != nil) != tt.shouldPanic {
+					t.Errorf("poolPathWithFeeDivide() panic = %v, shouldPanic = %v", r != nil, tt.shouldPanic)
+				}
+			}()
+
+			token0, token1, fee := poolPathWithFeeDivide(tt.input)
+			if !tt.shouldPanic {
+				if token0 != tt.wantToken0 {
+					t.Errorf("token0 = %v, want %v", token0, tt.wantToken0)
+				}
+				if token1 != tt.wantToken1 {
+					t.Errorf("token1 = %v, want %v", token1, tt.wantToken1)
+				}
+				if fee != tt.wantFee {
+					t.Errorf("fee = %v, want %v", fee, tt.wantFee)
+				}
+			}
+		})
+	}
+}
+
+func TestGetDataForSinglePath(t *testing.T) {
+	tests := []poolPathWithFeeDivideTestCases{
+		{
+			name:        "valid path",
+			input:       "tokenA:tokenB:500",
+			wantToken0:  "tokenA",
+			wantToken1:  "tokenB",
+			wantFee:     int(500),
+			shouldPanic: false,
+		},
+		{
+			name:        "invalid path format",
+			input:       "tokenA:tokenB",
+			shouldPanic: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer func() {
+				r := recover()
+				if (r != nil) != tt.shouldPanic {
```

Comment:
use uassert.PanicsWithMessage to check panic with panic messsage

Reply:
this function is unused. removed https://github.com/gnoswap-labs/gnoswap/pull/431/commits/80468af96073f7240198d2e874a193b14a890228